NASA-IMPACT/csdap-cumulus
# Integrated Hygen with Cumulus, configured a rule generator for making…

Integrated hygen.io with Cumulus, configured a rule generator for making rule files and provided examples.
#123

To test this:

// Pull down the latest repository information
```
git fetch
git pull
```


// Checkout the working branch
```
git checkout iss123_integrating_hygen_io
```


// Enter the Docker Container (There is a new node lib in here so if something isn't right, you may have to rebuild the image 
although that should happen automatically)
```
make bash
```

// Enter a command to make a single rule file (using command line parameters)
```	
hygen makeRule new_rule --start_date_YYYY 2006 --dataset_name WV0X_MSI_L1B --version_str 2 --provider_path_format 'css/nga/WV03/1B/'yyyy/DDD
```
Example:
```
	(csda-app-sbx-7894:kris-sbx7894):/work $ hygen makeRule new_rule --start_date_YYYY 2006 --dataset_name WV0X_MSI_L1B --version_str 2 --provider_path_format 'css/nga/WV03/1B/'yyyy/DDD
		OUTPUT:
			Loaded templates: _templates
			       added: app/stacks/cumulus/resources/rules/WV0X_MSI_L1B/WV0X_MSI_L1B___2_2006.json
```

// Enter a command to make a dataset (using prompts)
```
hygen makeRule new_rule
```
	Example:
```
	(csda-app-sbx-7894:kris-sbx7894):/work $ hygen makeRule new_rule
		OUTPUT (with prompt lines):
			(csda-app-sbx-7894:kris-sbx7894):/work $ hygen makeRule new_rule
			✔ Start Year (Format: YYYY): · 2001
			✔ Dataset Name (Example: WV04_MSI_L1B): · WV0Y_MSI_L1B
			✔ Version Number (Example: 1): · 3
			✔ providerPathFormat (Example: 'css/nga/WV03/1B/'yyyy/DDD): · 'css/nga/WV0Y/1B/'yyyy/DDD

			Loaded templates: _templates
			       added: app/stacks/cumulus/resources/rules/WV0Y_MSI_L1B/WV0Y_MSI_L1B___3_2001.json
			(csda-app-sbx-7894:kris-sbx7894):/work $ 
```

// For reference, see the example output that was commited
The template with variables is defined at:
```
	_templates/makeRule/new_rule/rule.ejs.t
```
The variables by themselves are defined at:
```
	_templates/makeRule/new_rule/prompt.js
```
Example Command with Commited output
```
	hygen makeRule new_rule --start_date_YYYY 2015 --dataset_name WV0X_HYGEN_TEST --version_str 3 --provider_path_format 'css/nga/WV0X/1B/'yyyy/DDD
```
Output from example command
```
	app/stacks/cumulus/resources/rules/WV0X_HYGEN_TEST/WV0X_HYGEN_TEST___3_2015.json
```
